src/features/user/DonorReceipt/microComponents/DonorDetails.tsx

```diff
@@ -0,0 +1,66 @@
+import type { AddressView, DonorView } from '../donorReceipt';
+import type { CountryCode } from '@planet-sdk/common';
+
+import { useMemo } from 'react';
+import { useTranslations } from 'next-intl';
+import styles from '../donationReceipt.module.scss';
+import { getFormattedAddress } from '../../../../utils/addressManagement';
+
+interface Props {
+  donor: DonorView | undefined;
+  address: AddressView | undefined;
+}
+
+const DonorDetails = ({ donor, address }: Props) => {
+  if (address === undefined || donor === undefined) return null;
+
+  const t = useTranslations('Donate');
+  const tCountry = useTranslations('Country');
+  const { country, zipCode, city, address1, address2 } = address;
+  const { type, name, tin } = donor;
+  const countryName = tCountry(country.toLowerCase() as Lowercase<CountryCode>);
+
+  const cityStatePostalString = useMemo(
+    () => getFormattedAddress(zipCode, city, null, countryName),
+    [zipCode, city, countryName]
+  );
+
+  return (
+    <div className={styles.donorDetails}>
+      <h3 className={styles.header}>
+        {t('donationReceipt.recipientInfoHeader')}
+      </h3>
+      <div className={styles.details}>
+        <div className={styles.donorName}>
+          <span className={styles.header}>
+            {t('donationReceipt.name', {
+              type,
+            })}
+          </span>
+          <span>{name}</span>
+        </div>
+        {tin && (
```

Comment:
Avoid using this pattern. It is easy to have leaked renders.

If using && to conditionally render, make sure the left side is a boolean value.